routes/auth/auth.routes.js

```diff
@@ -0,0 +1,141 @@
+const User = require('../../models/User.model');
+
+const router = require('express').Router();
+const mongoose = require('mongoose');
+
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+
+const bcryptjs = require('bcryptjs');
+const saltRounds = 10;
+
+// require auth middleware
+const { isLoggedIn, isLoggedOut } = require('../../middleware/middleware.js');
+
+// https://github.com/howardmann/authentication
+passport.use(
+  'local-login',
+  new LocalStrategy(
+    {
+      // Fields to accept
+      usernameField: 'username',
+      passwordField: 'password',
+      passReqToCallback: true, // allows us to access req in the call back
+    },
+    async (req, username, password, done) => {
+      username = req.sanitize(username);
+      password = req.sanitize(password);
+      // Check if user and password is valid
+      let user = await User.findOne({ username });
+      let passwordValid = user && bcryptjs.compareSync(password, user.passwordHash);
```

Comment:
never compareSync! 
this hashes the provided plain text password –> hence is slow. Node's way of having only 1 main thread will cause problems here. Namely because compareSync will block that thread. if you have a couple hundred users per minute this will severely impact server response times.